In [1]:
import numpy as np

1. Найти собственные векторы и собственные значения для линейного оператора, заданного матрицей

$$A=\begin{pmatrix}
-1 &amp; -6\\
2 &amp; 6
\end{pmatrix}.$$
Решение:

$$\begin{vmatrix}
-1-\lambda &amp; -6\\
2 &amp; 6-\lambda
\end{vmatrix} = 0$$
$(-1 - \lambda)(6 - \lambda)-(2\cdot -6) = 0$

$\lambda^{2} - 5 \cdot \lambda + 6 = 0,$

$D = 5^{2} - 4\cdot 1 \cdot 6=1$

$\lambda=(5\pm1)/2$

$\lambda1=6/2=3$

$\lambda2=4/2=2$

Найдем собственные векторы для $\lambda1$ система уравнений будет выглядеть:

$$\begin{cases}
-x_{1}-6\cdot x_{2}= 2\cdot x_{1}, \\
2\cdot x_{1}+6\cdot x_{2}= 2\cdot x_{2}.
\end{cases}$$
Перенеся все в левую часть, получим

$$\begin{cases}
-3\cdot x_{1}-6\cdot x_{2}= 0, \\
2\cdot x_{1}+4\cdot x_{2}= 0;
\end{cases}$$$$\begin{cases}
x_{2}=-\frac{1}{2}\cdot x_{1}, \\
0 = 0.
\end{cases}$$
Найдем собственные векторы для $\lambda2$ система уравнений будет выглядеть:

$$\begin{cases}
-x_{1}-6\cdot x_{2}= 3\cdot x_{1}, \\
2\cdot x_{1}+6\cdot x_{2}= 3\cdot x_{2}.
\end{cases}$$
Перенеся все в левую часть, получим

$$\begin{cases}
-4\cdot x_{1}-6\cdot x_{2}= 0, \\
2\cdot x_{1}+3\cdot x_{2}= 0;
\end{cases}$$$$\begin{cases}
x_{2}=-\frac{2}{3}\cdot x_{1}, \\
0 = 0.
\end{cases}$$

In [2]:
a = np.array([[-1, -6], [2, 6]])
w, v = np.linalg.eig(a)

print(f'Матрица A:\n{a}')
print(f'Собственные значения:\n{w}')
print(f'Собственные векторы:\n{v}')

Матрица A:
[[-1 -6]
 [ 2  6]]
Собственные значения:
[2. 3.]
Собственные векторы:
[[-0.89442719  0.83205029]
 [ 0.4472136  -0.5547002 ]]


Матрица A:
[[-1 -6]
 [ 2  6]]
Собственные значения:
[2. 3.]
Собственные векторы:
[[-0.89442719  0.83205029]
 [ 0.4472136  -0.5547002 ]]
2. Дан оператор поворота на 180 градусов, задаваемый матрицей

$$A=\begin{pmatrix}
-1 &amp; 0\\
0 &amp; -1
\end{pmatrix}.$$
Показать, что любой вектор является для него собственным.

Решение:

$$\begin{vmatrix}
-1-\lambda &amp; 0\\
0 &amp; -1-\lambda
\end{vmatrix} = 0$$
$(-1\cdot \lambda)^{2} = 0,$

$\lambda^{2} + 2\lambda + 1 = 0,$

$D = 2^{2} - 4\cdot 1 \cdot 1 = 0$

$\lambda=-2/2=-1$

Теперь найдем собственные векторы вида $\begin{pmatrix}
x_{1}\\
x_{2}
\end{pmatrix}$, подставив полученное собственное значение в выражение $\textbf{A}x=\lambda x$:

$$\begin{pmatrix}
-1 &amp; 0\\
0 &amp; -1
\end{pmatrix}
\begin{pmatrix}
x_{1}\\
x_{2}
\end{pmatrix}=
\lambda
\begin{pmatrix}
x_{1}\\
x_{2}
\end{pmatrix}.
$$
Получим:

$$\begin{cases}
-1\cdot x_{1} + 0\cdot x_{2} = -1\cdot x_{1}, \\
0\cdot x_{1} + -1\cdot x_{2}= -1\cdot x_{2}.
\end{cases}$$
В итоге получим систему следующего вида:

$$\begin{cases}
-x_{1} = -x_{1}, \\
-x_{2} = -x_{2};
\end{cases}$$$$\begin{cases}
0 = 0, \\
0 = 0.
\end{cases}$$
Следовательно, любой вектор является для данного оператора поворота будет собственным

3. Пусть линейный оператор задан матрицей

$$A=\begin{pmatrix}
1 &amp; 1\\
-1 &amp; 3
\end{pmatrix}.$$
Установить, является ли вектор $x=(1,1)$ собственным вектором этого линейного оператора.

Предположим, что вектор $x$ является собственным вектором заданного линейного оператора, тогда должно существовать некоторое вещественное число $\lambda$, при котором

$$\begin{pmatrix}
1 &amp; 1\\
-1 &amp; 3
\end{pmatrix}
\begin{pmatrix}
1\\
1
\end{pmatrix}=
\lambda
\begin{pmatrix}
1\\
1
\end{pmatrix}.
$$
Из этого будет следовать, что

$$\begin{cases}
1+1=\lambda \\
-1+3 = \lambda
\end{cases}
\Rightarrow
\begin{cases}
\lambda=2\\
\lambda=2
\end{cases}.$$
Следовательно, вектор $x=(1,1)$ является собственным вектором линейного оператора, заданного матрицей $A$.

4. Пусть линейный оператор задан матрицей

$$A=\begin{pmatrix}
0 &amp; 3 &amp; 0\\
3 &amp; 0 &amp; 0\\
0 &amp; 0 &amp; 3
\end{pmatrix}.$$
Установить, является ли вектор $x=(3, -3, -4)$ собственным вектором этого линейного оператора.

Предположим, что вектор $x$ является собственным вектором заданного линейного оператора. Тогда получим:

$$\begin{pmatrix}
0 &amp; 3 &amp; 0\\
3 &amp; 0 &amp; 0\\
0 &amp; 0 &amp; 3
\end{pmatrix}
\begin{pmatrix}
3\\
-3\\
-4
\end{pmatrix}=
\lambda
\begin{pmatrix}
3\\
-3\\
-4
\end{pmatrix}
$$$$\Rightarrow
\begin{cases}
\lambda=-3\\
\lambda=-3\\
\lambda=3
\end{cases}$$
Такая система не имеет смысла, следовательно, вектор $x=(3, -3, -4)$ не является собственным вектором линейного оператора, заданного матрицей $A$